graph/resolver.go

```diff
@@ -0,0 +1,151 @@
+package graph
+
+import (
+	"database/sql"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/RoongJin/pokedex-graphql-sqlite/graph/model"
+)
+
+// This file will not be regenerated automatically.
+//
+// It serves as dependency injection for your app, add any dependencies you require here.
+
+type Database struct {
+	DBPointer *sql.DB
+}
+
+type Resolver struct {
+	DB Database
+}
+
+func checkErr(err error) {
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+func AddPokemon(db *sql.DB, name string, description string, category string, typeOf string, abilities string) int64 {
+	stmt, err := db.Prepare("INSERT INTO Pokemons(Name, Description, Category, Type, Abilities) values(?,?,?,?,?)")
+	checkErr(err)
+
+	res, err := stmt.Exec(name, description, category, typeOf, abilities)
+	checkErr(err)
+
+	id, err := res.LastInsertId()
+	checkErr(err)
+	fmt.Println(id)
+
+	defer stmt.Close()
+	return id
+}
+
+func UpdatePokemon(db *sql.DB, id int, name string, description string, category string, typeOf string, abilities string) int64 {
+	stmt, err := db.Prepare("update Pokemons set Name=?, Description=?, Category=?, Type=?, Abilities=? where ID=?")
+	checkErr(err)
+
+	res, err := stmt.Exec(name, description, category, typeOf, abilities, id)
+	checkErr(err)
+
+	affected, err := res.RowsAffected()
+	checkErr(err)
+
+	fmt.Println(affected)
+
+	defer stmt.Close()
+	return affected
+}
+
+func DeletePokemon(db *sql.DB, id int) int64 {
+	stmt, err := db.Prepare("delete from Pokemons where ID=?")
+	checkErr(err)
+
+	res, err := stmt.Exec(id)
+	checkErr(err)
+
+	affected, err := res.RowsAffected()
+	checkErr(err)
+
+	fmt.Println(affected)
+
+	defer stmt.Close()
+	return affected
+}
+
+func GetAllPokemons(db *sql.DB) ([]*model.Pokemon, error) {
+	rows, err := db.Query("select * from Pokemons")
+	checkErr(err)
```

Comment:
always return error to outside, let the caller decide what to do with the error:

best practice for handling errors
```
if err != nil {
   return nil, err
}
```